snapcraft/application.py

```diff
@@ -121,6 +121,17 @@ def __init__(self, *args, **kwargs):
         # Whether we know that we should use the core24-based codepath.
         self._known_core24 = False
 
+        env_vars = {
+            ev: "SNAP" + ev for ev in ("CRAFT_BUILD_FOR", "CRAFT_BUILD_ENVIRONMENT")
+        }
+        for new_envvar, old_envvar in env_vars.items():
```

Comment:
```suggestion
        for craft_var, snapcraft_var in env_vars.items():
```

I think these names are more descriptive with regards to the source of the variables